core/arch/riscv/kernel/asm-defines.c

```diff
@@ -6,6 +6,10 @@
 #include <gen-asm-defines.h>
 #include <kernel/thread.h>
 #include <kernel/thread_private.h>
+#include <gen-asm-defines.h>
+#include <kernel/boot.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
```

Comment:
`<gen-asm-defines.h>` is already included. I suspect you need only need to add `<mm/core_mmu_arch.h>`